arraycontext/impl/pyopencl/fake_numpy.py

```diff
@@ -208,6 +210,31 @@ def all(self, a):
             result = result.get()[()]
         return result
 
+    def array_equal(self, a, b):
```

Comment:
@kaushikcfd What do you think this should return? Device scalar or "just plain `bool`"?

Reply:
For what it's worth, the other reductions (e.g. `any`) seem to return a device scalar if given arrays and a host scalar if given host scalars.

Reply:
I agree with @alexfikl, we should probably avoid implicit freeze-ing as much as possible.